enigma/src/main/java/cuchaz/enigma/utils/Utils.java

```diff
@@ -51,6 +43,19 @@ public static void delete(Path path) throws IOException {
         }
     }
 
+    public static String getFileExtension(Path path) {
+        Path name = path.getFileName();
+
+        // null for empty paths and root-only paths
+        if (name == null) {
+            return "";
+        }
+
+        String fileName = name.toString();
+        int dotIndex = fileName.lastIndexOf('.');
+        return dotIndex == -1 ? "" : fileName.substring(dotIndex + 1);
```

Comment:
I don't think this handles files or directories starting with a dot and without an extension